### PR TITLE
feat(clients/java): added TLS support in Java client

### LINKED ISSUE #2837: Add TLS support in the Java client
### Objective
The goal of this issue is to support TLS in the Java client.

### Expected Behavior 
This feature should be disabled by default and the user should be able to configure the client to enable or disable it, for example: 

```
    final ZeebeClient unsecureClient =
        ZeebeClient.newClientBuilder().brokerContactPoint(broker).usePlaintext().build(); // default

    final ZeebeClient secureClient =
        ZeebeClient.newClientBuilder().brokerContactPoint(broker).useTls().build();
```

Blocked by #2836

### PR DESCRIPTION
**Pull request by miguelpires**
## Description
Adds TLS support to Java client

## Related issues
closes #2837 

## Pull Request Checklist

- [x] I have signed the [Camunda CLA](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#sign-the-contributor-license-agreement)
- [x] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [x] If submitting code, please run `mvn clean install -DskipTests` locally before committing

### CODE REVIEW

File: clients/java/src/main/java/io/zeebe/client/impl/ZeebeClientImpl.java [hunk 1]
```diff
@@ -103,10 +103,16 @@ public static ManagedChannel buildChannel(ZeebeClientConfiguration config) {
...
 
-    // TODO: Issue #1134 - https://github.com/zeebe-io/zeebe/issues/1134
-    return ManagedChannelBuilder.forAddress(address.getHost(), address.getPort())
-        .usePlaintext()
-        .build();
+    final NettyChannelBuilder channelBuilder =
+        NettyChannelBuilder.forAddress(address.getHost(), address.getPort());
+
+    if (config.isSecureConnectionEnabled()) {
+      channelBuilder.useTransportSecurity().sslContext(config.getSslContext());
```
Comment (npepinpe) [inline clients/java/src/main/java/io/zeebe/client/impl/ZeebeClientImpl.java]: If `config.getSslContext()` returns `null`, does still still work as expected?

File: qa/integration-tests/src/test/java/io/zeebe/broker/it/SecurityTest.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.RuleChain;
+import org.junit.rules.Timeout;
+
+public class SecurityTest {
+  public Timeout testTimeout = Timeout.seconds(120);
+
+  public ClusteringRule clusteringRule =
+      new ClusteringRule(
```
Comment (npepinpe) [inline qa/integration-tests/src/test/java/io/zeebe/broker/it/SecurityTest.java]: Do we need to use 3 brokers? Only 1 makes the test a little bit faster, less setup.

File: qa/integration-tests/src/test/java/io/zeebe/broker/it/SecurityTest.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  private SslContext createContext() {
+    try {
+      return GrpcSslContexts.forClient()
+          .trustManager(
+              io.zeebe.broker.it.clustering.DeploymentClusteredTest.class
+                  .getClassLoader()
+                  .getResourceAsStream("security/ca.cert.pem"))
+          .build();
+    } catch (SSLException e) {
+      Loggers.SYSTEM_LOGGER.warn("Expected to create SSL context during test but failed with: ", e);
```
Comment (npepinpe) [inline qa/integration-tests/src/test/java/io/zeebe/broker/it/SecurityTest.java]: Why not let the exception bubble up? It's a test, and we probably want to fail the test if we didn't get the context as expected?

**Review by npepinpe — COMMENTED** [review]
:+1: 

Minor stuff in tests, and just one question.

File: clients/java/src/main/java/io/zeebe/client/impl/ZeebeClientImpl.java (hunk 1 above)
Comment (MiguelPires) [reply clients/java/src/main/java/io/zeebe/client/impl/ZeebeClientImpl.java]: Yes, in that case the default value is used

**Comment by MiguelPires** [comment]
Thanks for the review @npepinpe. I improved the test as you mentioned

File: clients/java/src/main/java/io/zeebe/client/impl/ZeebeClientImpl.java (hunk 1 above)
Comment (npepinpe) [reply clients/java/src/main/java/io/zeebe/client/impl/ZeebeClientImpl.java]: :+1:

**Review by npepinpe — APPROVED** [review]

**Comment by MiguelPires** [comment]
bors retry

**Pull request merged**